flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/checkpointing/SingleCheckpointBarrierHandler.java

```diff
@@ -257,8 +257,8 @@ private void abortInternal(long cancelledId, CheckpointException exception) thro
                 Math.max(lastCancelledOrCompletedCheckpointId, cancelledId);
         numBarriersReceived = 0;
         controller.abortPendingCheckpoint(cancelledId, exception);
-        allBarriersReceivedFuture.completeExceptionally(exception);
         notifyAbort(cancelledId, exception);
+        allBarriersReceivedFuture.completeExceptionally(exception);
```

Comment:
I think this line change should be in another commit compared with below changes to ensure exception is not overridden by `CHECKPOINT_ASYNC_EXCEPTION`.

Reply:
The commit to avoid exception not overridden by `CHECKPOINT_ASYNC_EXCEPTION` should also be picked into release-1.11 and release-1.12.

Reply:
I'd think that we should backport all fixes in this commit, so I'd be okay with leaving it as is.

Reply:
Ok, I will split those two changes.